elastic/integrations
# Make K8s integration asset centric

- Enhancement

## Proposed commit message

WHAT: We change the structrure of the Kubernetes Integration ,trying to make it more resource centric. This means that the grouping of the configuration options for kubernetes collection (metrics or logss) is done based on the resource type that users want to monitor.  This done with such a way in order to help UX and become more user friendly. 

WHY:
See more info in https://github.com/elastic/integrations/issues/7936


## Checklist



## How to test this PR locally

- Clone dir
- cd packages/integrations/kubernetes
- elastic-package build
- elastic-package stack up -d -vvv --version=8.14.0-SNAPSHOT  
- Create an elastic-agent policy
- Add kubernetes integration. The version 1.58.0 should be available

### Testing upgrade procedure

I have also tested the upgrade procedure from an old k8s integration to this new structure

You can follow the steps:
1. Initially go and add an older version of the integration in your policy by using link
`https://localhost:5601/app/fleet/integrations/kubernetes-1.55.0/add-integration`
2. Install the elastic-agent
3. Go to Stack Management> Index Management > Data Streams
Choose Hidden datastreams

![Screenshot 2024-03-13 at 5 24 05 PM](https://github.com/elastic/integrations/assets/1685415/1456a2fa-01e7-4c1e-a4b3-2198c1e6579c)


4. Then you can try upgrading to v1.58.0 
Choose the policy and click kubernetes option
![Screenshot 2024-03-13 at 5 27 12 PM](https://github.com/elastic/integrations/assets/1685415/e150555a-91ad-4a1e-b4a5-95030af6c07d)

5. Go to Settings and choose to upgrade to latest version
![Screenshot 2024-03-13 at 5 27 31 PM](https://github.com/elastic/integrations/assets/1685415/8679a3a6-bf1a-49cd-9c4b-95e5266ef422)

**6. Upgrade was successful and also the logs/metrics collection continued successfully**

## Related issues

- Closes #https://github.com/elastic/integrations/issues/7936

## Screenshots

How the full list of the datastream option looks like: 
![fulllist](https://github.com/elastic/integrations/assets/1685415/1d7e0140-3c4b-4b8a-98dd-9ef35a578967)


We group Cluster Resource Metrics
![cluster_resource](https://github.com/elastic/integrations/assets/1685415/ffee27c6-b331-4b87-90c8-f4bea5bf9dbd)

Example of Pod Metrics

![Pod_metrics](https://github.com/elastic/integrations/assets/1685415/bcaca2e4-a976-46f5-9277-caef232b8dbd)


Small video for demonstration for UI teams:

https://github.com/elastic/integrations/assets/1685415/b733f00e-83a1-4e96-8bc6-60a8a333e286
